chains/solana/contracts/programs/cctp-token-pool/src/context.rs

```diff
@@ -0,0 +1,739 @@
+        bump,
+        seeds::program = cctp_token_messenger_minter,
+    )]
+    pub cctp_remote_token_messenger_key: UncheckedAccount<'info>,
+
+    /// CHECK this is the account in which CCTP will store the event. It is not a PDA of CCTP,
+    /// but CCTP will initialize it and become the owner for it.
+    #[account(
+        mut,
+        seeds = [
+            MESSAGE_SENT_EVENT_SEED,
+            &lock_or_burn.original_sender.to_bytes(),
+            &lock_or_burn.remote_chain_selector.to_le_bytes(),
+            &lock_or_burn.msg_total_nonce.to_le_bytes(),
+        ],
+        bump,
+        owner = System::id(), // this is not initialized yet, it will later be owned by CCTP
+    )]
+    pub cctp_message_sent_event: UncheckedAccount<'info>,
+}
+
+#[derive(Accounts)]
+#[instruction(remote_chain_selector: u64, mint: Pubkey)]
+pub struct InitializeChainConfig<'info> {
+    #[account(
+        seeds = [
+            POOL_STATE_SEED,
+            mint.key().as_ref()
+        ],
+        bump,
+    )]
+    pub state: Account<'info, State>,
+
+    #[account(
+        init,
+        seeds = [
+            POOL_CHAINCONFIG_SEED,
+            &remote_chain_selector.to_le_bytes(),
+            mint.key().as_ref(),
+        ],
+        bump,
+        payer = authority,
+        space = ANCHOR_DISCRIMINATOR + ChainConfig::INIT_SPACE,
+    )]
+    pub chain_config: Account<'info, ChainConfig>,
+
+    #[account(mut, address = state.config.owner)]
+    pub authority: Signer<'info>,
+
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+#[instruction(remote_chain_selector: u64, mint: Pubkey)]
+pub struct EditChainConfig<'info> {
+    #[account(
+        seeds = [
+            POOL_STATE_SEED,
+            mint.key().as_ref()
+        ],
+        bump,
+    )]
+    pub state: Account<'info, State>,
+
+    #[account(
+        mut,
+        seeds = [
+            POOL_CHAINCONFIG_SEED,
+            &remote_chain_selector.to_le_bytes(),
+            mint.key().as_ref(),
+        ],
+        bump,
+    )]
+    pub chain_config: Account<'info, ChainConfig>,
+
+    #[account(mut, constraint = authority.key() == state.config.owner || authority.key() == state.config.rate_limit_admin)]
+    pub authority: Signer<'info>,
+}
+
+#[derive(Accounts)]
+#[instruction(remote_chain_selector: u64, mint: Pubkey, cfg: RemoteConfig)]
+pub struct EditChainConfigDynamicSize<'info> {
+    #[account(
+        seeds = [
+            POOL_STATE_SEED,
+            mint.key().as_ref()
+        ],
+        bump,
+    )]
+    pub state: Account<'info, State>,
+
+    #[account(
+        mut,
+        seeds = [
+            POOL_CHAINCONFIG_SEED,
+            &remote_chain_selector.to_le_bytes(),
+            mint.key().as_ref(),
+        ],
+        bump,
+        realloc = ANCHOR_DISCRIMINATOR + ChainConfig::INIT_SPACE + cfg.pool_addresses.iter().map(RemoteAddress::space).sum::<usize>(),
```

Comment:
Not unique to this, but I wonder if there's a cleaner way to represent these `realloc` size calculations that is resilient to us adding new dynamic fields to the struct. Something like borsh-serializing the incoming `cfg` and measuring the size diff with the INIT one, or something like that.